### PR TITLE
AA-1705 | Multiple AuthStrategies (Backend & UI Implementations)

### PR DESCRIPTION
**Pull request by lchavarriaga**
Implementations of AA1705 & AA1711

### CODE REVIEW

**Review by CSR2017 — COMMENTED** [review]
@lchavarriaga 
Please fix all the warnings. On the UI - Auth strategy override pop up page, the auth strategy names are not readable/ full text is not visible, is there any way of fixing that ?

File: Application/EdFi.Ods.AdminApp.Management/ClaimSetEditor/ClaimSetResourceClaimActionAuthStrategies.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace EdFi.Ods.AdminApp.Management.ClaimSetEditor;
+
+public interface IClaimSetResourceClaimActionAuthStrategies
+{
+    IList<AuthorizationStrategy?> AuthorizationStrategies { get; }
+}
```
Comment (stephenfuqua) [inline Application/EdFi.Ods.AdminApp.Management/ClaimSetEditor/ClaimSetResourceClaimActionAuthStrategies.cs:13]: Please resolve these warning before trying to merge this.

File: Application/EdFi.Ods.AdminApp.Management/ClaimSetEditor/GetResourcesByClaimSetIdQueryV53Service.cs [hunk 2]
```diff
@@ -27,7 +28,7 @@ public GetResourcesByClaimSetIdQueryV53Service(ISecurityContext securityContext,
             _mapper = mapper;
         }
 
-        internal void AddChildResourcesToParents(IEnumerable<ResourceClaim> childResources, List<ResourceClaim> parentResources)
+        internal static void AddChildResourcesToParents(IReadOnlyList<ResourceClaim> childResources, IList<ResourceClaim> parentResources)
         {
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply Application/EdFi.Ods.AdminApp.Management/ClaimSetEditor/GetResourcesByClaimSetIdQueryV53Service.cs:32]: If it compiles, then making it static is the right thing to do.

**Review by stephenfuqua — COMMENTED** [review]
Thank you for the updates on my comments. I leave the rest of the review up to the team.

**Review by CSR2017 — APPROVED** [review]

File: Application/EdFi.Ods.AdminApp.Management/ClaimSetEditor/GetResourcesByClaimSetIdQuery.cs [hunk 3]
```diff
@@ -23,24 +23,24 @@ public class GetResourcesByClaimSetIdQuery : IGetResourcesByClaimSetIdQuery
...
         {
-            List<ResourceClaim> parentResources;
+            IList<ResourceClaim> parentResources;
             var securityModel = _resolver.DetermineSecurityModel();
             if (securityModel == EdFiOdsSecurityModelCompatibility.ThreeThroughFive)
             {
                 parentResources = _v53Service.GetParentResources(claimSetId);
                 var childResources = _v53Service.GetChildResources(claimSetId);
-                _v53Service.AddChildResourcesToParents(childResources, parentResources);
+                GetResourcesByClaimSetIdQueryV53Service.AddChildResourcesToParents(childResources, parentResources);
```
Comment (stephenfuqua) [inline Application/EdFi.Ods.AdminApp.Management/ClaimSetEditor/GetResourcesByClaimSetIdQuery.cs:34]: @lchavarriaga can you help me understand why this changed from using the class variable, whose value was set through dependency injection, to using a static method? Doesn't this make unit testing much harder?

Comment (stephenfuqua) [reply Application/EdFi.Ods.AdminApp.Management/ClaimSetEditor/GetResourcesByClaimSetIdQuery.cs:34]: Never mind - I see that the `AddChildResourcestoParents` method does not rely on any state information in its class, and therefore static was actually more appropriate. Muy bien.

**Pull request merged**